### PR TITLE
Minor fix

### PR DESCRIPTION
**Pull request by weihuoya**
~~remove ".rar", ".zip", ".7z", ".torrent", ".tar", ".gz" in allowed extensions;~~
call getWidth and getHeight before bitmap be recycled;
better user experience with fade icon;
fix icon data size;

### CODE REVIEW

**Comment by BreadFish64** [comment]
To elaborate, we want users to know that compressed files will not work, rather than silently hiding them.

**Comment by saintbepis** [comment]
> To elaborate, we want users to know that compressed files will not work, rather than silently hiding them.

Wouldn't it be better to make it an option and leave it on by default for those that have too many things in the folders that hold their games?

File: src/android/app/src/main/java/org/citra/citra_emu/model/GameDatabase.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public void scanLibrary(SQLiteDatabase database) {
                 null);    // Order of folders is irrelevant.
 
         Set<String> allowedExtensions = new HashSet<String>(Arrays.asList(
-                ".3ds", ".3dsx", ".elf", ".axf", ".cci", ".cxi", ".app", ".rar", ".zip", ".7z", ".torrent", ".tar", ".gz"));
```
Comment (bunnei) [inline src/android/app/src/main/java/org/citra/citra_emu/model/GameDatabase.java:145]: This is actually intentional -- we show common unsupported file types so users can see that they have the right directory, but that citra does not support these file types. Can you revert this part of the change? Thank you!

**Comment by bunnei** [comment]
LGTM, thank you!

**Comment by bunnei** [comment]
@weihuoya also did you see that https://github.com/citra-emu/citra-android/pull/92 was reverted? This was due to the UI being less responsive when running below fullspeed. Please consider fixing the issue and resubmitting the PR. Thank you!

**Pull request merged**